elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/util/impl/ElasticsearchDateHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.util.impl;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * Various utilities to manipulate dates and comply with Elasticsearch date format.
+ *
+ * @author Guillaume Smet
+ */
+public final class ElasticsearchDateHelper {
+
+	private static final TimeZone ENCODING_TIME_ZONE = TimeZone.getTimeZone( "UTC" );
```

Comment:
Mmmh, no, in fact, it's voluntary.

From the Es doc: "In JSON documents, dates are represented as strings. Elasticsearch uses a set of preconfigured formats to recognize and parse these strings into a long value representing milliseconds-since-the-epoch in UTC.".

In practice, it's the same but we might as well be in line with the doc.